packages/@aws-cdk/core/lib/custom-resource-provider/cross-region-ssm-writer-handler/index.ts

```diff
@@ -0,0 +1,103 @@
+/*eslint-disable no-console*/
+/* eslint-disable import/no-extraneous-dependencies */
+import { SSM } from 'aws-sdk';
+type CrossRegionExports = { [exportName: string]: string };
+
+export async function handler(event: AWSLambda.CloudFormationCustomResourceEvent) {
+  const props = event.ResourceProperties;
+  const exports: CrossRegionExports = props.Exports;
+
+  const ssm = new SSM({ region: props.Region });
+  try {
+    switch (event.RequestType) {
+      case 'Create':
+        console.info(`Creating new SSM Parameter exports in region ${props.Region}`);
+        await throwIfAnyInUse(ssm, exports);
+        await putParameters(ssm, exports);
+        return;
+      case 'Update':
+        const oldProps = event.OldResourceProperties;
+        const oldExports: CrossRegionExports = oldProps.Exports;
+        const newExports = filterExports(exports, oldExports);
+        await throwIfAnyInUse(ssm, newExports);
+        console.info(`Creating new SSM Parameter exports in region ${props.Region}`);
+        await putParameters(ssm, newExports);
+        return;
+      case 'Delete':
+        // consuming stack will delete parameters
```

Comment:
It should be safe to delete parameters that are not in use, right? I'm afraid we might leak them otherwise.

Reply:
We would leak them until the consuming stack is deleted since the consuming stack will delete all of it's parameters (by path prefix). As it is we wouldn't have a list of all parameters that this resource has ever created, just what it had on the last update. We could store the state somewhere (another parameter?).

Reply:
We know which references are strong or weak beforehand, so we can apply different rules for them.

Right now, if the producing stack fails to create for some reason and rolls back, we never get to deploy the consumer and we will leak.